salt/states/boto_ec2.py

```diff
@@ -169,6 +169,8 @@ def eni_present(
         description=None,
         groups=None,
         source_dest_check=True,
+        allocate_eip=False,
```

Comment:
Region, key, keyid, and profile are never expected to be passed positionally. My preference is for new args to be before region.